lib/liquid/standardfilters.rb

```diff
@@ -194,8 +194,12 @@ def map(input, property)
         if property == "to_liquid".freeze
           e
         elsif e.respond_to?(:[])
-          r = e[property]
-          r.is_a?(Proc) ? r.call : r
+          begin
+            r = e[property]
+            r.is_a?(Proc) ? r.call : r
+          rescue TypeError
```

Comment:
Hmm I wonder if there's another way to catch this that doesn't involve a `rescue`.

Reply:
I discussed this with Sam, and `e[property]` must be called in order to verify its own existence. Otherwise, checking if `e` has attribute `property` beforehand requires knowledge of `property` type (e.g.; if `e` is an `Array` and `property` is a `string`, then by calling `e.[](property)` we pass in a string argument, which causes an error).